lib/test/integration/cli.go

```diff
@@ -12,153 +12,64 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package e2e
+package integration
 
 import (
 	"bytes"
 	"fmt"
-	"io"
 	"os/exec"
 	"strings"
-	"testing"
 
 	"github.com/pkg/errors"
 )
 
-type kn struct {
-	namespace string
-}
-
 const (
 	seperatorHeavy = "━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━"
 	seperatorLight = "╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍"
 )
 
-// Run the 'kn' CLI with args and opts
-func (k kn) Run(args ...string) KnRunResult {
-	return RunKn(k.namespace, args)
-}
-
-// Helper methods for calling out to the test cluster
-type kubectl struct {
+type Kn struct {
 	namespace string
 }
 
-// Run the 'kubectl' CLI with args and opts
-func (k kubectl) Run(args ...string) (string, error) {
-	return RunKubectl(k.namespace, args...)
+// New Kn object
+func NewKn() Kn {
```

Comment:
@maximilien : looking at equivalent kubectl definition taking namespace args, while `NewKn` isnt taking any. 
```
func NewKubectl(namespace string) Kubectl {
```

@rhuss : also noticed similar difference for `Run` method of `kn` and `kubectl`; one requires explicit namespace arg while another takes it from object. We can align it to use namespace set in each CLI's object ? 
As all the calls to these CLIs in a test should operate in same namespace? or I am missing something here ?

Reply:
@navidshaikh afair there where certain usage of kubectl without namespace (would have to check the usage), so haven't put it into the struct itself. Let me check that later ...

Reply:
Ah, its the way around. Actually, I'm not sure if the `NewKubectl` or `NewKn` methods are really necessary as in many other spots `Kn{}` or `Kubectl{}`  is used directly. We should not mix and at the moment I don't see any benefit for a constructor method.

Reply:
> as in many other spots Kn{} or Kubectl{} is used directly. We should not mix and at the moment I don't see any benefit for a constructor method.

+1

Reply:
Cool!

In that case, lets remove the constructor methods, initialize `Kn{}` or `Kubectl{}` and add namespace as required.

Reply:
Can't use `Kn{namespace}` or `Kubectl{namespace}` if the struct field is private. And would be a bad idea to make it public since everything depends on namespace and should not expose this... it really should not change. So my vote is to only have the `NewKn` and `NewKubectl` functions only.